### PR TITLE
Login page, initial routing and markup for main page

### PR DESCRIPTION
**Pull request by NessInesska**

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -1,16 +1,41 @@
...
+import { appRoutes } from './app.routes';
+import { MainPageComponent } from './components/main-page';
+import { AuthGuard } from './guards';
+import { ProductCardComponent } from './components/product-card';
 
 @NgModule({
   declarations: [
-    AppComponent
+    AppComponent,
+    AuthPageComponent,
```
Comment (dmitriirudakov) [inline src/app/app.module.ts]: Module for each page

File: src/app/components/auth-page/auth-page.component.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    passwordInput: ['', Validators.required],
+  });
+
+  constructor(private authorizationService: AuthorizationService,
+              private formBuild: FormBuilder,
+              private routingService: RoutingService,
+              private userService: UserService) {};
+
+  public onLogin() {
+    let username = this.loginForm.controls['loginInput'].value;
```
Comment (dmitriirudakov) [inline src/app/components/auth-page/auth-page.component.ts]: username, login, currentUser...

File: src/app/components/auth-page/auth-page.component.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    } else {
+      this.authorizationService.login(username, password)
+        .subscribe(
+          token => this.authorizationService.setToken(token));
+      this.routingService.goToMainPage();
+      this.userService.currentUser = username;
+    }
+  }
+
+    public get f() {
```
Comment (dmitriirudakov) [inline src/app/components/auth-page/auth-page.component.ts]: f - bad name

File: src/app/constants/route.constants.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import {environment} from '../../environments/environment';
+
+const API_PREFIX='/api';
+
+export const ROUTING_PATHES = {
+  LOGIN: `${environment.baseUrl}${API_PREFIX}/login`,
```
Comment (dmitriirudakov) [inline src/app/constants/route.constants.ts]: why routes have api prefix??

File: src/app/guards/main-page-guard.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import {ActivatedRouteSnapshot, CanActivate, RouterStateSnapshot} from '@angular/router';
+
+import {RoutingService, UserService} from '../services';
+
+@Injectable()
+export class AuthGuard implements CanActivate{
```
Comment (dmitriirudakov) [inline src/app/guards/main-page-guard.ts]: two AuthGuard ?

File: src/app/components/main-page/main-page.component.scss [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    order: 1;
+    width: 100vw;
+    padding: 20px;
+    background: black;
+    color: $white;
+    margin: 0;
+  }
+}
+
+* {
```
Comment (dmitriirudakov) [inline src/app/components/main-page/main-page.component.scss]: it very very very common style. why is it living in one of component styles?

File: src/app/components/auth-page/auth-page.component.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+              private routingService: RoutingService,
+              private userService: UserService) {};
+
+  public onLogin() {
+    let username = this.loginForm.controls['loginInput'].value;
+    let password = this.loginForm.controls['passwordInput'].value;
+
+    if(this.loginForm.invalid) {
+      return;
+    } else {
```
Comment (dmitriirudakov) [inline src/app/components/auth-page/auth-page.component.ts]: else block can be skipped as you already did return

File: src/app/app.module.ts (hunk 1 above)
Comment (NessInesska) [reply src/app/app.module.ts]: done

File: src/app/components/auth-page/auth-page.component.ts (hunk 3 above)
Comment (NessInesska) [reply src/app/components/auth-page/auth-page.component.ts]: done

File: src/app/constants/route.constants.ts (hunk 4 above)
Comment (NessInesska) [reply src/app/constants/route.constants.ts]: done

File: src/app/components/auth-page/auth-page.component.ts (hunk 7 above)
Comment (NessInesska) [reply src/app/components/auth-page/auth-page.component.ts]: done

File: src/app/guards/main-page-guard.ts (hunk 5 above)
Comment (NessInesska) [reply src/app/guards/main-page-guard.ts]: done

File: src/app/components/main-page/main-page.component.scss (hunk 6 above)
Comment (NessInesska) [reply src/app/components/main-page/main-page.component.scss]: done

File: src/app/components/auth-page/auth-page.component.ts (hunk 2 above)
Comment (NessInesska) [reply src/app/components/auth-page/auth-page.component.ts]: done

File: src/app/components/main-page/main-page.component.scss [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+      min-height: fit-content;
+      min-width: 200px;
+      box-shadow: 0 0 40px -10px rgba(0, 0, 0, 0.75);
+    }
+  }
+
+  .header {
+    flex-shrink: 0;
+    padding: 20px;
+    background: black;
```
Comment (NessInesska) [inline src/app/components/main-page/main-page.component.scss]: move to styles/colors.scss

File: src/app/components/main-page/main-page.component.scss [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    min-width: 300px;
+  }
+
+  .footer {
+    display: flex;
+    align-content: center;
+    order: 1;
+    width: 100vw;
+    padding: 20px;
+    background: black;
```
Comment (NessInesska) [inline src/app/components/main-page/main-page.component.scss]: move to styles/colors.scss

File: src/app/services/routing.service.ts [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RoutingService {
+
+  constructor(private router: Router) { }
+
+  public goToMainPage() {
+    this.router.navigate(['http://localhost:3000/api/main']);
```
Comment (NessInesska) [inline src/app/services/routing.service.ts]: fix link

File: src/app/styles/colors.scss [hunk 11]
```diff
@@ -0,0 +1,2 @@
+$white: #fff;
+$login-body-background: #f3eeee;
```
Comment (NessInesska) [inline src/app/styles/colors.scss]: move colors to constants

File: src/app/app.module.ts [hunk 12]
```diff
@@ -1,16 +1,40 @@
...
+import { RouterModule } from '@angular/router';
 
 import { AppComponent } from './app.component';
+import { appRoutes } from './app.routes';
+import { AuthPageModule } from './components/auth-page';
+import { ProductCardModule } from './components/product-card';
+import { AuthGuard } from './guards';
+import { MainPageModule } from './components/main-page';
+import { AuthorizationService, RoutingService, UserService } from './services';
+import { ProductCardService } from './services/product-card.service';
```
Comment (NessInesska) [inline src/app/app.module.ts]: barrel

**Pull request merged**